### PR TITLE
Added some more client methods and examples

### PR DESCRIPTION
**Pull request by leighwaller**

### CODE REVIEW

File: bin/example.rb [hunk 1]
```diff
@@ -42,36 +43,36 @@ def send_message
...
+  puts "Executing send multiple messages..."
+
+  client = Messagemedia::SOAP::Client.new(USER_ID, PASSWORD)
+
+  # Construct the message
+  message1 = Messagemedia::SOAP::Message.new
+  message1.content = "Content of Message 1 to Recipient 1"
+  message1.delivery_report = true
+  message1.format = Messagemedia::SOAP::FORMAT_SMS
+  message1.validity_period = 1
```
Comment (afmm) [inline bin/example.rb]: I know this was what we had before, but what time unit is this? Would it be better to set the demo to a more realistic value?

File: bin/example.rb [hunk 2]
```diff
@@ -97,32 +98,32 @@ def check_user_info
...
-            puts "Reply origin: #{reply[:origin]}"
-            puts "Reply content: #{reply[:content]}"
-            puts "Reply format: #{reply[:'@format']}"
-            receipt_ids.push reply[:'@receipt_id']
-        end
+  if not replies[:replies].nil? then
+    replies[:replies].each do |reply|
+      puts "Reply receipt id: #{reply[:'@receipt_id']}"
+      puts "Reply uid: #{reply[:'@uid']}"
+      puts "Reply received now time: #{reply[:received]}"
```
Comment (afmm) [inline bin/example.rb]: This doesn't sound right to me, perhaps just "Reply received time"?

File: lib/messagemedia/soap/message.rb [hunk 3]
```diff
@@ -10,7 +10,7 @@ module SOAP
     #
     class Message
       attr_accessor :sequence_number, :origin, :recipients, :content,
-                    :validity_period, :format, :delivery_report
+                    :validity_period, :format, :delivery_report, :scheduled
```
Comment (afmm) [inline lib/messagemedia/soap/message.rb]: I don't know Ruby, but I assume this wont break backwards compatibility?

File: bin/example.rb (hunk 1 above)
Comment (Ox5f3759df) [inline bin/example.rb]: We default the validity period to 169 in the Corporate Gateway. Set it to that.

File: bin/example.rb [hunk 4]
```diff
@@ -130,33 +131,107 @@ def check_and_confirm_replies
...
+def delete_scheduled_message
+  puts "Scheduling message..."
+
+  client = Messagemedia::SOAP::Client.new(USER_ID, PASSWORD)
+
+  # Construct the message
+  message = Messagemedia::SOAP::Message.new
+  message.content = "Content of Message"
+  message.format = Messagemedia::SOAP::FORMAT_SMS
+  message.validity_period = 1
```
Comment (Ox5f3759df) [inline bin/example.rb]: Change validity period here as well.

File: lib/messagemedia/soap/message.rb (hunk 3 above)
Comment (Ox5f3759df) [inline lib/messagemedia/soap/message.rb]: i think line 76 covers the event in which scheduled isn't set i think.

File: bin/example.rb (hunk 1 above)
Comment (leighwaller) [inline bin/example.rb]: From the doco...

(Optional) This element specifies the validity period of the message.
Should the message-recipient be unavailable the service provider will
continue to attempt delivery until the delivery period expires.
Validity period only applies to SMS messages. If this element is
specified the value must be an unsigned byte (i.e. an integer between 0
and 255 inclusive). The value of the validityPeriod element
determines the validity time period as follows:
 For values ranging from 0 to 143 the time period is equal to:
(value + 1) x 5 minutes
 For values ranging from 144 to 167 the time period is equal to:
12 hours + (value – 143) x 30 minutes
 For values ranging from 168 to 196 the time period is equal to:
(value – 166) x 1 day
 For values ranging from 197 to 255 the time period is equal to:
(value – 192) x 1 week
If this element is not specified a value of 169 is assumed which equates
to 3 days

Comment (leighwaller) [inline bin/example.rb]: so if im reading this correctly a validityPeriod of 1 would keep alive for 10 mins. Is that not realistic?

File: bin/example.rb (hunk 2 above)
Comment (leighwaller) [inline bin/example.rb]: yeh well spotted...i did a replace all date to now :-1:

File: bin/example.rb (hunk 1 above)
Comment (afmm) [inline bin/example.rb]: That sounds fine (thumbsup)

File: lib/messagemedia/soap/message.rb (hunk 3 above)
Comment (leighwaller) [inline lib/messagemedia/soap/message.rb]: Thats just setting getters for various fields

**Pull request merged**